File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isValidBST(TreeNode root) {
        if(root == null) return true;
        boolean l = true, r = true;
        if(root.left != null){
            TreeNode temp = root.left;
            while(temp.right != null) temp = temp.right;
            if(temp.val < root.val) l = isValidBST(root.left);
            else return false;
        }
        if(root.right != null){
            TreeNode temp = root.right;
            while(temp.left != null) temp = temp.left;
            if(temp.val > root.val) r = isValidBST(root.right);
            else return false;
        }
        return l&r;
    }
}

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void getIn(TreeNode* root, vector<int> &inorder){
        if(!root) return;
        getIn(root->left, inorder);
        inorder.push_back(root->val);
        getIn(root->right, inorder);
    }
    bool isValidBST(TreeNode* root) {
        vector<int> inorder;
        getIn(root, inorder);
        for(int i=0; i<inorder.size()-1; i++){
            if(inorder[i] >=  inorder[i+1]) return 0;
        }
        return 1;
    }
};